renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -37,13 +37,410 @@ constexpr VkDeviceSize handleCountSize = 8;
+            continue;
+          }
+        }
+
+        BufferData transformData;
+        if(triInfo.transformData.deviceAddress != 0x0)
+        {
+          transformData = GetDeviceAddressData(triInfo.transformData.deviceAddress);
+          if(!transformData)
+          {
+            RDCERR("Unable to find VkBuffer for transform data at 0x%llx",
+                   triInfo.transformData.deviceAddress);
+            continue;
+          }
+        }
+
+        // Find the alignment requirements for each type
+        {
+          VkMemoryRequirements mrq = {};
+
+          // Vertex buffer.  The complexity here is that the rangeInfo members are interpreted
+          // differently depending on whether or not index buffers are used
+          ObjDisp(device)->GetBufferMemoryRequirements(Unwrap(device), vertexData.buf, &mrq);
+          vertexData.alignment = mrq.alignment;
+          vertexData.size = vertexData.buf != VK_NULL_HANDLE
+                                ? vertexData.rao.record->memSize - vertexData.rao.offset
+                                : rangeInfo.primitiveCount * 3 * triInfo.vertexStride;
+          vertexData.start =
+              indexData ? 0
+                        : rangeInfo.primitiveOffset + (triInfo.vertexStride * rangeInfo.firstVertex);
+
+          // Index buffer
+          if(indexData)
+          {
+            ObjDisp(device)->GetBufferMemoryRequirements(Unwrap(device), indexData.buf, &mrq);
+            indexData.alignment = mrq.alignment;
+            indexData.size = rangeInfo.primitiveCount * 3 * IndexTypeSize(triInfo.indexType);
+            indexData.start = rangeInfo.primitiveOffset;
+          }
+
+          // Transform buffer
+          if(transformData)
+          {
+            ObjDisp(device)->GetBufferMemoryRequirements(Unwrap(device), transformData.buf, &mrq);
+            transformData.alignment = mrq.alignment;
+            transformData.size = sizeof(VkTransformMatrixKHR);
+            transformData.start = rangeInfo.transformOffset;
+          }
+        }
+        const VkDeviceSize maxAlignment =
+            RDCMAX(RDCMAX(vertexData.alignment, indexData.alignment), transformData.alignment);
+
+        // We want to copy the ubput buffers into one big block so sum the sizes up together
+        const VkDeviceSize totalMemSize = AlignUp(vertexData.size, vertexData.alignment) +
+                                          AlignUp(indexData.size, indexData.alignment) +
+                                          AlignUp(transformData.size, transformData.alignment);
+
+        readbackmem = CreateReadBackMemory(device, totalMemSize, maxAlignment);
+        if(readbackmem.mem == VK_NULL_HANDLE)
+        {
+          RDCERR("Unable to allocate AS triangle input buffer readback memory (size: %u bytes)",
+                 totalMemSize);
+          continue;
+        }
+
+        // Insert copy commands
+        VkBufferCopy region = {
+            vertexData.rao.offset + vertexData.start,
```

Comment:
I think there's a risk of confusion here in future with the terms `offset` and `start` being somewhat interchangeable. Here `offset` meaning "offset from the beginning of the actual buffer to the address passed in" and `start` being "the relative location where the geometry data sits".

Since you are using the local `BufferData` you could maybe rename these variables and add a helper function for this, but I wonder if you need to keep `start` separate? It looks like everywhere you use this it's always added to offset since once we've processed all the offsets and sizes here we really only care about the absolute offset from buffer start to where the data sits, the address the application passed in is mostly irrelevant.